src/app/app/layout.tsx

```diff
@@ -14,15 +14,12 @@ export const metadata: Metadata = {
 export default async function RootLayout({ children }: PropsWithChildren) {
   const session = await getServerSession(authOptions);
 
-  if (!session) {
-    const origin = getOriginPath();
-    return redirect(makeRedirectURL(origin));
-  }
+  const loggedUser = await fetchUserWithSession(session);
 
   return (
     <AuthProvider>
       <header className="bg-white shadow">
-        <Navbar />
+        <Navbar loggedUser={loggedUser} />
```

Comment:
Lets create a task for moving this to context provider 🙏🏼

Reply:
I created that task on our Notion. I'll be working on that on my next PR.